tests/functional/openstackdataplanenodeset_controller_test.go

```diff
@@ -211,6 +211,140 @@ var _ = Describe("Dataplane NodeSet Test", func() {
+						CtlplaneInterface:     "",
+						CtlplaneGateway:       "",
+						CtlplaneNetmask:       "255.255.255.0",
+						BmhNamespace:          "openshift-machine-api",
+						HardwareReqs: baremetalv1.HardwareReqs{
+							CPUReqs: baremetalv1.CPUReqs{
+								Arch:     "",
+								CountReq: baremetalv1.CPUCountReq{Count: 0, ExactMatch: false},
+								MhzReq:   baremetalv1.CPUMhzReq{Mhz: 0, ExactMatch: false},
+							},
+							MemReqs: baremetalv1.MemReqs{
+								GbReq: baremetalv1.MemGbReq{Gb: 0, ExactMatch: false},
+							},
+							DiskReqs: baremetalv1.DiskReqs{
+								GbReq:  baremetalv1.DiskGbReq{Gb: 0, ExactMatch: false},
+								SSDReq: baremetalv1.DiskSSDReq{SSD: false, ExactMatch: false},
+							},
+						},
+						PasswordSecret:   nil,
+						CloudUserName:    "",
+						DomainName:       "",
+						BootstrapDNS:     nil,
+						DNSSearchDomains: nil,
+					},
+					NodeTemplate: dataplanev1.NodeTemplate{
+						AnsibleSSHPrivateKeySecret: "dataplane-ansible-ssh-private-key-secret",
+						Networks:                   nil,
+						ManagementNetwork:          "ctlplane",
+						Ansible: dataplanev1.AnsibleOpts{
+							AnsibleUser: "cloud-admin",
+							AnsibleHost: "",
+							AnsiblePort: 0,
+							AnsibleVars: nil,
+						},
+						ExtraMounts: nil,
+						UserData:    nil,
+						NetworkData: nil,
+					},
+					Env:                nil,
+					PreProvisioned:     true,
+					NetworkAttachments: nil,
+					TLSEnabled:         tlsEnabled,
+					Nodes:              map[string]dataplanev1.NodeSection{},
+					Services: []string{
+						"download-cache",
+						"bootstrap",
+						"configure-network",
+						"validate-network",
+						"install-os",
+						"configure-os",
+						"run-os",
+						"reboot-os",
+						"install-certs",
+						"ovn",
+						"neutron-metadata",
+						"libvirt",
+						"nova",
+						"telemetry",
+						"global-service"},
+				}
+				Expect(dataplaneNodeSetInstance.Spec).Should(Equal(emptyNodeSpec))
+			})
+
+			It("should have the Status fields initialized", func() {
+				dataplaneNodeSetInstance := GetDataplaneNodeSet(dataplaneNodeSetName)
+				Expect(dataplaneNodeSetInstance.Status.Deployed).Should(BeFalse())
+			})
+
+			It("should have input not ready and unknown Conditions initialized", func() {
+				th.ExpectCondition(
+					dataplaneNodeSetName,
+					ConditionGetterFunc(DataplaneConditionGetter),
+					condition.ReadyCondition,
+					corev1.ConditionFalse,
+				)
+				th.ExpectCondition(
+					dataplaneNodeSetName,
+					ConditionGetterFunc(DataplaneConditionGetter),
+					condition.InputReadyCondition,
+					corev1.ConditionFalse,
+				)
+				th.ExpectCondition(
+					dataplaneNodeSetName,
+					ConditionGetterFunc(DataplaneConditionGetter),
+					dataplanev1.SetupReadyCondition,
+					corev1.ConditionFalse,
+				)
+			})
+
+			It("Should not have created a Secret", func() {
+				th.AssertSecretDoesNotExist(dataplaneSecretName)
+			})
+		})
```

Comment:
could we have test that proves that AEE resource is reacted for the global service with the proper config?

Reply:
I'll put that among deployment functional tests. But I'm going to make an assertion here on the service itself.

Reply:
Test is now included in dataplanedeploymentcontroller suite.